toolchain/sem_ir/name_scope.h

```diff
@@ -60,19 +60,22 @@ class NameScope : public Printable<NameScope> {
     return lookup.value();
   }
 
-  // Adds a new name known to not exist. Must not be poisoned.
+  // Adds a new name known to not exist. The new entry may not be poisoned. This
+  // is allowed even if the name has already been poisoned.
```

Comment:
If the name was already poisoned, is the new entry still poisoned? That's particularly important to capture since we're looking at storing instructions for poisoned names, so "has instruction and is poisoned" is becoming a valid state. Maybe:

```suggestion
  // Adds a new name known to not exist. The new entry won't be poisoned, and
  // can overwrite poisoned names.
```

Reply:
Comment updated.